test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -242,6 +243,26 @@ class SyncConfigTests {
         assertEquals(config.partitionValue.asString(), partitionValue)
     }
 
+    @Test
+    fun name() {
```

Comment:
I think we lack some corner case tests here:

- E.g. partition-based default names. They should be something like `s_mypartition.realm` or `i_42.realm`.
- What happens if you bring a filename with no `.realm` suffix. 

Since our default name is `default.realm` and if ObjectStore doe append `.realm` to all names, then I could be tempted to also enforce it for local Realms, but that would be a breaking change. Not sure what your thoughts are?

It does have the advantage that you can always double-click on a file and get it to open in Realm Studio though.

Reply:
I think it would be fair to add the constraint for local Realms for the sake of uniformity.